tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -962,6 +962,67 @@ def delete_smartrest_one_template(self, template_id: str):
                 ex,
             )
 
+    @keyword("List Entities")
+    def list_entities(
+            self, name: str = None 
+    ) -> List[Dict[str, Any]]:
+        """
+        Get entity list from the device using the entity store query REST API
+
+        Args:
+            name (str, optional): Device name to fetch the entity list from
+
+        Returns:
+            List[Dict[str, Any]]: List of entities
+
+        *Example:*
+        | ${entities}= | List Entities |
+        | ${entities}= | List Entities | name=${PARENT_SN} |
+        """
+        device = self.current
+        if name:
+            if name in self.devices:
+                device = self.devices.get(name)
+
+        if not device:
+            log.info("Device has not been setup, so no logs to collect")
+            return
+
+        output = device.execute_command("curl -f http://localhost:8000/tedge/entity-store/v1/entities")
+        entities = json.loads(output.stdout)
+        return entities
+
+
+    @keyword("Assert Contains Entity")
```

Comment:
It's convention to use `Should Contain ...`in the keyword. It does not have to match the python method name.

```suggestion
    @keyword("Should Container Entity")
```